src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```
@@ -0,0 +1,64 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.CinemaHall;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(cinemaHall);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(
+                    "Can't add CinemaHall entity, error: ", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return cinemaHall;
+    }
+
+    @Override
+    public Optional<CinemaHall> get(Long id) {

```

This method signature is correct in declaring Optional<CinemaHall>, but the implementation below uses getSingleResult() which will throw an exception when no entity exists, causing your DAO to throw instead of returning Optional.empty(). This violates the checklist requirement: "DAO get(...) methods should return Optional<T> rather than null or throwing if not found (see checklist item 6)." Consider using session.get(CinemaHall.class, id) and returning Optional.ofNullable(result), or use Query.uniqueResultOptional() so absence is represented by Optional.empty().